core/src/main/java/org/elasticsearch/percolator/MultiDocumentPercolatorIndex.java

```diff
@@ -71,7 +72,16 @@ public void prepare(PercolateContext context, ParsedDocument parsedDocument) {
         try {
             MultiReader mReader = new MultiReader(memoryIndices, true);
             LeafReader slowReader = SlowCompositeReaderWrapper.wrap(mReader);
-            final IndexSearcher slowSearcher = new IndexSearcher(slowReader);
+            final IndexSearcher slowSearcher = new IndexSearcher(slowReader) {
+
+                @Override
+                public void search(Query query, Collector collector) throws IOException {
+                    BooleanQuery.Builder bq = new BooleanQuery.Builder();
+                    bq.add(query, BooleanClause.Occur.MUST);
+                    bq.add(Queries.newNestedFilter(), BooleanClause.Occur.MUST_NOT);
+                    super.search(bq.build(), collector);
```

Comment:
ok, it just caught my attention because if you call directly lower-level methods this could be by-passed. I guess it's fine for now, thanks for the explanation